sqlx-core/src/sqlite/options/connect.rs

```diff
@@ -18,22 +18,9 @@ impl ConnectOptions for SqliteConnectOptions {
             let mut conn = establish(self).await?;
 
             // send an initial sql statement comprised of options
-            //
-            // page_size must be set before any other action on the database.
-            //
-            // Note that locking_mode should be set before journal_mode; see
-            // https://www.sqlite.org/wal.html#use_of_wal_without_shared_memory .
-            let init = format!(
-                "PRAGMA page_size = {}; PRAGMA locking_mode = {}; PRAGMA journal_mode = {}; PRAGMA foreign_keys = {}; PRAGMA synchronous = {}; PRAGMA auto_vacuum = {}",
-                self.page_size,
-                self.locking_mode.as_str(),
-                self.journal_mode.as_str(),
-                if self.foreign_keys { "ON" } else { "OFF" },
-                self.synchronous.as_str(),
-                self.auto_vacuum.as_str(),
-            );
-
-            conn.execute(&*init).await?;
+            for (key, value) in self.pragmas.iter() {
+                conn.execute(&*format!("PRAGMA {} = {};", key, value)).await?;
+            }
```

Comment:
Since there is a bit of overhead in `.execute()` I think it would be better if we collected the `PRAGMA` statements together in one string and executed that.